### PR TITLE
Get the lastest version of verimag packages (camlp4 free)

### PR DESCRIPTION
**Pull request by jahierwan**

### CODE REVIEW

**Comment by mseri** [comment]
The findlib files can be removed, they had and have no effect.
The triples of `opam`, `url` and `descr` files need to be merge into single `opam` files in the `2.0` format. The `1.2` format is no longer supported and used in the `opam-repository`.

**Comment by mseri** [comment]
You can see what the new files look like by looking at the old `lustre` version (it was automatically converted when we upgraded to `opam2`): https://github.com/ocaml/opam-repository/blob/master/packages/lustre-v6/lustre-v6.1.737/opam

**Comment by mseri** [comment]
The documentation for opam2 and how to convert the opam files is https://opam.ocaml.org/doc/Upgrade_guide.html#What-you-need-to-be-aware-of_1

There is a plugin, `opam-package-upgrade`, that can do the conversion for you (I don't remember if it also moves the descriptions or not, you may need to do that by hand)

**Comment by jahierwan** [comment]
I will look at that. Thanks for your quick feedback.

**Comment by jahierwan** [comment]
> The documentation for opam2 and how to convert the opam files is https://opam.ocaml.org/doc/Upgrade_guide.html#What-you-need-to-be-aware-of_1
> 
> There is a plugin, `opam-package-upgrade`, that can do the conversion for you (I don't remember if it also moves the descriptions or not, you may need to do that by hand)

Indeed. My url and descr (and opam) files were automatically generated via oasis2opam... 
Doing it by hand is fine for this repo is ok, but for my local repo that is automatically updated, it is
a problem.  Is it ok for people that uses my repo if I continue to use opam 1.2 for a while ?

**Comment by mseri** [comment]
> Doing it by hand is fine for this repo is ok, but for my local repo that is automatically updated, it is
> a problem. Is it ok for people that uses my repo if I continue to use opam 1.2 for a while ?

Absolutely, that's not a problem

**Comment by mseri** [comment]
Github is telling me that there are conflicts with  
```
packages/lustre-v6/lustre-v6.1.737/opam
packages/lutin/Lutin.2.56/opam 
```
and the PR cannot be merged. Maybe it is easier to open a fresh PR or force push the final version on this branch

**Comment by jahierwan** [comment]
Marcello Seri <notifications@github.com> writes:

> Github is telling me that there are conflicts with
>
> packages/lustre-v6/lustre-v6.1.737/opam
> packages/lutin/Lutin.2.56/opam 
>
> and the PR cannot be merged. Maybe it is easier to open a fresh PR or force push the final version on this branch

I've seen this message too late. I've juste merge the upstream repo
with my fork. I hope it's ok.

**Comment by mseri** [comment]
Abslutely, no problem at all

File: packages/lustre-v6/lustre-v6.6.101.7/opam [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+homepage: "http://www-verimag.imag.fr/lustre-v6.html"
+bug-reports: "http://www-verimag.imag.fr/lustre-v6.html"
+depends: [
+  "ocaml" {>= "4.05"}
+  "base-unix"
+  "extlib" {build} | "extlib-compat" {build}
+  "lutils" {>= "1.49"}
+  "oasis" {build & >= "0.4.7"} | "oasis-mirage" {build & >= "0.4.7"}
+  "ocamlbuild" {build}
+  "ocamlfind"
```
Comment (mseri) [inline packages/lustre-v6/lustre-v6.6.101.7/opam]: Is this a build dependency or you depend on `findlib` ar runtime?

File: packages/lustre-v6/lustre-v6.6.101.7/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+]
+build: [
+  [make "build"]
+  ["oasis" "setup"] {with-test}
+  ["ocaml" "setup.ml" "-configure" "--enable-tests"] {with-test}
+  ["ocaml" "setup.ml" "-build"] {with-test}
+  ["ocaml" "setup.ml" "-test"] {with-test}
+]
+install: [make "install"]
+remove: [
```
Comment (mseri) [inline packages/lustre-v6/lustre-v6.6.101.7/opam]: This is no longer needed with `opam2`. It tracks the installed files and knows what to do with uninstall (similarly in the other opam files)

File: packages/lutils/lutils.1.49.5/opam [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+license: "CeCILL"
+homepage:
+  "https://gricad-gitlab.univ-grenoble-alpes.fr/verimag/synchrone/lutils/"
+bug-reports:
+  "https://gricad-gitlab.univ-grenoble-alpes.fr/verimag/synchrone/lutils/"
+depends: [
+  "ocaml" {>= "4.02"}
+  "base-unix"
+  "ocamlbuild" {build}
+  "ocamlfind"
```
Comment (mseri) [inline packages/lutils/lutils.1.49.5/opam]: Same as in the other file. If it is only needed at build, please add `{build}`

File: packages/lustre-v6/lustre-v6.6.101.7/opam (hunk 1 above)
Comment (jahierwan) [reply packages/lustre-v6/lustre-v6.6.101.7/opam]: I'm not 100% sure. Maybe it does as this package contains a dynamic lib (lustre-v6.cmxs).
This dependancy was generated by oasis2opam (which might not be a proof that it is
correct w.r.t. the need of findlib at runtime.

File: packages/lustre-v6/lustre-v6.6.101.7/opam (hunk 2 above)
Comment (jahierwan) [reply packages/lustre-v6/lustre-v6.6.101.7/opam]: ok, cool. I will remove it.

File: packages/lustre-v6/lustre-v6.6.101.7/opam (hunk 1 above)
Comment (mseri) [reply packages/lustre-v6/lustre-v6.6.101.7/opam]: No problem, let's keep it as it is

**Comment by mseri** [comment]
Thanks for the patience and the care! I would have not complained for the lint warnings :)

**Pull request merged**